Src/FluentAssertions.Net40/Xml/XmlReaderValidator.cs

```diff
@@ -0,0 +1,239 @@
+﻿using FluentAssertions.Execution;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml;
+
+namespace FluentAssertions.Xml
+{
+    class XmlReaderValidator
+    {
+        AssertionScope assertion;
+        XmlReader subjectReader, expectedReader;
+
+        string CurrentLocation
+        {
+            get { return "/" + string.Join("/", locationStack.Reverse()); }
+        }
+
+        Stack<string> locationStack = new Stack<string>();
+
+        public XmlReaderValidator(XmlReader subjectReader, XmlReader expectedReader, string because, object[] reasonArgs)
+        {
+            assertion = Execute.Assertion.BecauseOf(because, reasonArgs);
+        
+            this.subjectReader = subjectReader;
+            this.expectedReader = expectedReader;
+        }
+                
+        class ValidationResult
```

Comment:
Is there a simple way to limit the number of reported errors if using nested scopes? To make multiple error reports useful I'm also afraid some more complex logic will be needed to get things in sync when elements are found in one of the readers that are not present in the other. Think of `<xml><a/><b/></c></xml>` vs `<xml><b/></c></xml>`. Doing good reports on that would require detection that a is missing, but b and c are correct.